types/hapi__joi/index.d.ts

```diff
@@ -152,1245 +226,1757 @@ export interface DomainOptions {
+         * String comparisons are by default case insensitive,
+         * see `boolean.sensitive()` to change this behavior.
+         * @param values - strings, numbers or arrays of them
+         */
+        falsy(...values: Array<string | number>): this;
+
+        /**
+         * Allows the values provided to truthy and falsy as well as the "true" and "false" default conversion
+         * (when not in `strict()` mode) to be matched in a case insensitive manner.
+         */
+        sensitive(enabled?: boolean): this;
+
+        /**
+         * Allows for additional values to be considered valid booleans by converting them to true during validation.
+         * String comparisons are by default case insensitive, see `boolean.sensitive()` to change this behavior.
+         * @param values - strings, numbers or arrays of them
+         */
+        truthy(...values: Array<string | number>): this;
+    }
+
+    interface NumberSchema extends AnySchema {
+        /**
+         * Specifies that the value must be greater than limit.
+         * It can also be a reference to another field.
+         */
+        greater(limit: number | Reference): this;
+
+        /**
+         * Requires the number to be an integer (no floating point).
+         */
+        integer(): this;
+
+        /**
+         * Specifies that the value must be less than limit.
+         * It can also be a reference to another field.
+         */
+        less(limit: number | Reference): this;
+
+        /**
+         * Specifies the maximum value.
+         * It can also be a reference to another field.
+         */
+        max(limit: number | Reference): this;
+
+        /**
+         * Specifies the minimum value.
+         * It can also be a reference to another field.
+         */
+        min(limit: number | Reference): this;
+
+        /**
+         * Specifies that the value must be a multiple of base.
+         */
+        multiple(base: number | Reference): this;
+
+        /**
+         * Requires the number to be negative.
+         */
+        negative(): this;
+
+        /**
+         * Requires the number to be a TCP port, so between 0 and 65535.
+         */
+        port(): this;
+
+        /**
+         * Requires the number to be positive.
+         */
+        positive(): this;
+
+        /**
+         * Specifies the maximum number of decimal places where:
+         * @param limit - the maximum number of decimal places allowed.
+         */
+        precision(limit: number): this;
+
+        /**
+         * Requires the number to be negative or positive.
+         */
+        sign(sign: 'positive' | 'negative'): this;
+
+        /**
+         * Allows the number to be outside of JavaScript's safety range (Number.MIN_SAFE_INTEGER & Number.MAX_SAFE_INTEGER).
+         */
+        unsafe(enabled?: any): this;
+    }
+
+    interface StringSchema extends AnySchema {
+        /**
+         * Requires the string value to only cont
```

Comment:
`coerce` seems to be `coerce?(schema: ExtensionBoundSchema, value: any, helpers:CustomHelpers): any;` now according to https://hapi.dev/family/joi/?v=16.0.1#extensions

Reply:
Sorry, having a second look and using the library I think the example is wrong and it's now:
```
    interface ErrorReport extends Error {
        code: string;
        flags: Record<string, Flag>;
        path: string;
        prefs: ValidationOptions;
        messages: LanguageMessages;
        state: State;
        value: any;
    }

    type CoerceFunction = (value: any, helpers:CustomHelpers) => {
        errors?:ErrorReport[]
        value?: any
    } | undefined

        coerce?: {
            from: string | string[],
            method: CoerceFunction
        } | CoerceFunction;
```

See:
https://github.com/hapijs/joi/blob/aa374d7034d8fccc166ad4a80601ce1ddd7ed8db/lib/extend.js#L61-L71  https://github.com/hapijs/joi/blob/bb1b40559f7c9b6abd62c70bcd5c49a5ca4951fe/lib/validator.js#L218-L232
https://github.com/hapijs/joi/blob/14efd782935a56e6b5631fd5fcf37958b570254c/lib/errors.js#L18-L24